.github/workflows/trivy-security-scan.yml

```diff
@@ -39,47 +39,93 @@ jobs:
           echo "::set-output name=available::true"
         shell: bash
 
-  trivy_scan_image:
+  update-trivy-db:
     needs: wait-for-image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup oras
+        uses: oras-project/setup-oras@9c92598691bfef1424de2f8fae81941568f5889c # v1.21
+
+      - name: Get current date
+        id: date
+        run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+
+      - name: Download and extract the vulnerability DB
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/.cache/trivy/db
+          oras pull ghcr.io/aquasecurity/trivy-db:2
+          tar -xzf db.tar.gz -C $GITHUB_WORKSPACE/.cache/trivy/db
+          rm db.tar.gz
+
+      - name: Cache DBs
+        uses: actions/cache@v4
+        with:
+          path: ${{ github.workspace }}/.cache/trivy
+          key: cache-trivy-${{ steps.date.outputs.date }}
+
+  trivy_scan_image:
+    needs: update-trivy-db
```

Comment:
This will result in update happening every single time, essentially defeating the purpose of caching, in fact caching just does more work here for no reason.

What should be done instead is a separate workflow like described in https://github.com/aquasecurity/trivy-action?tab=readme-ov-file#updating-caches-in-the-default-branch that updates cache periodically and then use it in this workflow.

Even better approach and something I'd do would be to only update the cache if it is missing and then if cache recovery (for current date) is successful, we'll skip downloading. I have implemented this logic for GTK4 building [here](https://github.com/autonomys/space-acres/blob/40deff6c5f2ca5cd8902481baaa1e8f2a4783ba8/.github/workflows/release.yml#L51-L71), the only major difference here is that the cache key will include a date.